pkg/checksec/canary.go

```diff
@@ -19,6 +19,50 @@ type CanaryResult struct {
 // StackChk to check for stack_chk_fail value
 const StackChk = "__stack_chk_fail"
 
+// validatePath ensures the path is within the current working directory
+func validatePath(path string) error {
+	if path == "" {
+		return fmt.Errorf("path is empty")
+	}
+
+	// Disallow absolute paths
+	if filepath.IsAbs(path) {
+		return fmt.Errorf("absolute paths are not allowed")
+	}
+
+	// Disallow any ".." component (before or after Clean)
+	if strings.Contains(path, "..") {
+		return fmt.Errorf("path traversal detected")
+	}
+
+	cleanPath := filepath.Clean(path)
+
+	// Double-check: ensure no ".." remains (defensive)
+	if strings.Contains(cleanPath, "..") {
```

Comment:
I'm very sorry. 
Missing import of "strings" package.

Please add:
import "strings"

to the top of the file, otherwise the code will fail to compile.
This was caused by my carelessness.